packages/fuel-gauge/src/token-test-contract.test.ts

```diff
@@ -141,4 +145,18 @@ describe('TokenTestContract', () => {
     // the contract should now have only 50 coins
     expect((await getBalance()).toHex()).toEqual(bn(50).toHex());
   });
+
+  it('throws when passing entire Address object as address parameter', async () => {
+    const userWallet = Wallet.generate({ provider });
+    const token = await setup();
+    const addressParameter = {
+      value: userWallet.address,
+    };
+    const assetId = BaseAssetId;
+
+    await expectToThrowFuelError(
+      () => token.functions.transfer_coins_to_output(addressParameter, assetId, 50).call(),
+      new FuelError(ErrorCode.ENCODE_ERROR, 'Invalid b256.')
+    );
+  });
```

Comment:
New test case to detail that we can no longer pass in an `Address` type intended to represent a `b256` address.